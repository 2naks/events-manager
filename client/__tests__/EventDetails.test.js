import React, { Components } from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import ConnectedEventDetails, { EventDetails } from '../components/EventDetails.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('EventDetails Component', () => {

  const initialState = {
    eventReducer: {
      status: "Error",
      message: "",
      event:{
        id: '1',
        title: 'Seminar',
        description: "",
        location: "",
        time:"5:00",
        date:Date.now(),
        image: "",
        Center:[]
      }
    }
  };

  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => {
      return str;
    }
  }

  const props = {
    dispatch: () => {},
    centerId: "1",
    event: initialState.eventReducer.event,
    match: {
      params: {
        id:'1'
      }
    },
  };

  describe('EventDetails form  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<EventDetails {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render connected component', () => {
      const wrapper = mount(<Provider store={store}><ConnectedEventDetails match={props.match}/></Provider>);
    });

  });

});
