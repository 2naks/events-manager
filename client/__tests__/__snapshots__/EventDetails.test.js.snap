// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetails Component EventDetails form  should render self it should render for dumb component 1`] = `
<div>
  <NavBar
    page="MyEvents"
  />
  <div
    className="container event-details"
  >
    <div
      className="row"
    >
      <div
        className="col-md-8 event"
      >
        <div
          className="card mb-3"
        >
          <img
            alt="Card image cap"
            className="card-img-top img-fluid"
            height="300"
            src=""
          />
          <div
            className="card-block"
          >
            <h5
              className="card-title"
            >
              <b>
                Seminar
              </b>
            </h5>
            <p
              className="card-text"
            />
          </div>
        </div>
        <Location
          location="undefined, undefined"
        />
      </div>
      <div
        className="col-md-4 venue-date"
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="card"
            >
              <h6
                className="card-header"
              >
                <b>
                  VENUE
                </b>
              </h6>
              <div
                className="card-block"
              >
                <p
                  className="card-text"
                >
                  undefined, undefined
                </p>
              </div>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="card"
            >
              <h6
                className="card-header"
              >
                <b>
                  DATE & TIME
                </b>
              </h6>
              <div
                className="card-block"
              >
                <p
                  className="card-text"
                >
                  01 January 2020
                </p>
                <p
                  className="card-text"
                >
                  5:00
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
